### ISSUE TITLE
Universe error in IDE mode

### ISSUE THREAD

**Issue by aseemr** [FStarLang/FStar#1448]
Jay Bosamiya reports the following example that fails in the IDE mode:

```
module Bug

abstract type seq (a:Type u#a) :Type u#a =
  | MkSeq: l:list a -> seq a
```

with:

```
(Error) ASSERTION FAILURE: Impossible: missing universe instantiation on length
F* may be in an inconsistent state.
Please file a bug report, ideally with a minimized version of the program that triggered the error.
```

Also heads up @cpitclaudel.

**Comment by aseemr** [comment]
IDE mode always fails for explicit universe annotations, here is an even simpler example:

```
val foo (a:Type u#a) :Type u#a
```

Here is the reason:

Recently I added code to generalize annotated universes in `ToSyntax.fs`. So, the desugaring of the `val` above is supposed to be:

```
val foo<a> :a:Type (u@0) -> Type (u@1)
```

i.e. the universe `a` should be generalized and then all its occurrences should become de-bruijn. I added it so that the typechecker can now treat it as usual (this was done as part of 2-phase changes).

However, I added this generalization under the function `desugar_module_common`, which does not get called in the IDE mode for every decl (duh!). And so, the typechecker raises a legit error.

I will try adding this generalization code directly to `desugar_decl`. Hopefully that should fix it.
